src/backend/gporca/libgpopt/src/xforms/CXformFactory.cpp

```diff
@@ -353,13 +353,11 @@ CXformFactory::IsXformIdUsed(CXform::EXformId exfid)
 //		Initializes global instance
 //
 //---------------------------------------------------------------------------
-GPOS_RESULT
+void
 CXformFactory::Init()
 {
 	GPOS_ASSERT(NULL == Pxff() && "Xform factory was already initialized");
 
-	GPOS_RESULT eres = GPOS_OK;
-
```

Comment:
> I think the ctors of all classes, related to this patch should be refactored and use the unified approach like it's done at CMessageRepository - there is catching and rethrowing of exception (it seems redundant) , instead of it used CAutoMemorryPool with further detach, so seems it may be rewritten like:

But I think that we need to get rid of CAutoMemoryPool during initialization.

Reply:
> I think mixing the Pxff() method (which returns pointer to static m_pxff) and acessing m_pxff explicitly is misleading,
> so assert above may be rewritten like `GPOS_ASSERT(NULL == m_pxff ...)`

Yes, but here it is done to reduce the diff.

Reply:
> GPOS_ASSERT(NULL == m_pxff ...)

done